### PR TITLE
Implement message removal flow

### PR DESCRIPTION
**Pull request by Vladyslav-Kuksiuk**
This PR adds an implementation of the message removal flow.

Message removal flow model:
image

### CODE REVIEW

File: server/src/main/java/io/spine/examples/chatspn/server/message/MessageAggregate.java [hunk 1]
```diff
@@ -99,4 +102,33 @@ MessageContentUpdated handle(UpdateMessageContent c) throws MessageContentCannot
...
+
+    /**
+     * Handles the command to mark a message as removed.
+     *
+     * @throws MessageCannotBeMarkedAsRemoved
+     *         if the message does not exist, or is already marked as removed
+     */
+    @Assign
+    MessageMarkedAsRemoved handle(MarkMessageAsRemoved c) throws MessageCannotBeMarkedAsRemoved {
+        if (!state().hasId() || state().getIsRemoved()) {
```
Comment (Artem-Semenov-dev) [inline server/src/main/java/io/spine/examples/chatspn/server/message/MessageAggregate.java:114]: `getIsRemoved()` sounds weird to me. Consider renaming.

**Review by Artem-Semenov-dev — COMMENTED** [review]
Almost LGTM, please see my comment.

File: model/src/main/proto/spine_examples/chatspn/message/message.proto [hunk 2]
```diff
@@ -56,6 +56,9 @@ message Message {
 
     // Time when this message was posted.
     google.protobuf.Timestamp when_posted = 5 [(required) = true];
+
+    // Answers the question 'Is the message removed?'.
+    bool is_removed = 6 [(required) = true];
```
Comment (armiol) [inline model/src/main/proto/spine_examples/chatspn/message/message.proto:61]: Can we just mark this aggregate as "deleted" instead?

File: model/src/main/proto/spine_examples/chatspn/message/removal_events.proto [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+    MessageId id = 1;
+
+    // The ID of the chat in which the message was removed.
+    ChatId chat = 2 [(required) = true];
+
+    // The ID of the user who removed the message.
+    spine.core.UserId user = 3 [(required) = true];
+}
+
+// The message failed to be removed.
```
Comment (armiol) [inline model/src/main/proto/spine_examples/chatspn/message/removal_events.proto:53]: Let's describe somewhere (maybe it's here?) what are the reasons for that. 

Right now there is no documentation explaining who can remove messages. Maybe, that's a command-level documentation. If we had such a place, we could reference it from the documentation of all related events and rejections.

**Review by armiol — CHANGES_REQUESTED** [review]
Please see my comments.

**Pull request merged**